src/include/stir/listmode/CListModeDataECAT.h

```diff
@@ -97,10 +97,11 @@ class CListModeDataECAT : public CListModeData
   mutable std::vector<std::vector<std::streampos> > saved_get_positions_for_each_lm_data;
   typedef std::pair<unsigned int, SavedPosition> GetPosition;
   std::vector<GetPosition > saved_get_positions;
-  
+
   // const as it modifies only mutable elements
   // It has to be const as e.g. get_next_record calls it
   Succeeded open_lm_file(unsigned int) const; 
+  shared_ptr<Scanner> scanner_sptr;
```

Comment:
a bit worrying. we now have a `proj_data_info` already. that has a `Scanner`. do we need this?

Reply:
This was tricky. There are some checks in the CListModeDataECAT.cxx on the scanner type and at line 82 it even goes ahead to set the scanner type. This would be dangerous with the ProjDatInfo. I removed that option, assuming that ProjDataInfo will be properly initialised.